### PR TITLE
Parallelize CPU tests to have them run faster.

### PR DESCRIPTION
**Pull request by hertschuh**
We are now using self-hosted CPU runners which are much more powerfully (16 cores), but we haven't been taking advantage of them. We can parallelize tests on CPU (but not on GPU or TPU) to make them 4 times faster overall (17 minutes instead of 64 minutes).

This will free up enough of the CPU runners to use them for KerasHub.

Test | Before | After
----|-----|-----
Tests / Run tests on CPU (jax) | 64m | 17m
Tests / Run tests on CPU (numpy) | 14m | 8m
Tests / Run tests on CPU (openvino) | 21m | 9m
Tests / Run tests on CPU (tensorflow) | 39m | 13m 
Tests / Run tests on CPU (torch) | 29m | 16m

Other changes:
- Carved out the SKLearn wrappers test, they use some weird parameterization scheme that doesn't play nice with pytest. This code is never modified anyway.
- Excluded some RandAugment test that is flaky on OpenVino
- Tweaked tolerance on some EinsumDense tests that were flaky on Torch

## Contributor Agreement
**Please check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: requirements-common.txt [hunk 1]
```diff
@@ -17,7 +16,9 @@ tensorboard-plugin-profile
 rich
 build
 optree
+pytest
 pytest-cov
+pytest-xdist
```
[the remark this answers is not in the document]
Comment (hertschuh) [reply requirements-common.txt:21]: It does

**Review by sachinprasadhs — APPROVED** [review]
